LayoutTest/TestCase/LYTLayoutFailingTestSnapshotRecorder.h

```diff
@@ -0,0 +1,27 @@
+// © 2016 LinkedIn Corp. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+
+@interface LYTLayoutFailingTestSnapshotRecorder : NSObject<XCTestObservation>
+
+@property (nonatomic, strong) UIView *viewUnderTest;
```

Comment:
It seems a little strange to have the singleton hold onto state for a specific test (though can see how it's necessary).

In the future, people may want to use this without an LYTLayoutTestCase, so I think we should try to anticipate that? Maybe holding onto them inside the test case makes the most sense given this? So no-one else will ever overwrite them by mistake?

Thoughts?